src/services/notifications.rs

```diff
@@ -41,35 +40,68 @@ pub async fn post_registration(
     let client = context.client();
     let mut requests = Vec::with_capacity(registration_request.safe_registrations.len());
 
-    for safe_registration in registration_request.safe_registrations.into_iter() {
+    for safe_registration in registration_request.safe_registrations.iter() {
         let info_provider = DefaultInfoProvider::new(&safe_registration.chain_id, &context);
         let url = core_uri!(info_provider, "/v1/notifications/devices/")?;
         let backend_request =
             build_backend_request(&registration_request.device_data, safe_registration);
 
-        requests.push(
+        requests.push((
+            &safe_registration.chain_id,
             client
                 .post(url.to_string())
                 .json(&backend_request)
                 .timeout(Duration::from_millis(default_request_timeout()))
                 .send(),
-        );
+        ));
     }
 
-    for request in requests.into_iter() {
-        request.await?;
-    }
+    let chain_id_errors = {
+        let mut output: Vec<&str> = vec![];
+
+        for (chain_id, request) in requests.into_iter() {
+            if let Ok(response) = request.await {
+                // tx service errors
+                if let Err(_) = forward_error(response).await {
+                    output.push(chain_id);
+                }
+            } else {
+                // reqwest failures (unreachable urls, do we need to handle?)
+                output.push(chain_id);
+            };
+        }
+        output
+    };
 
-    Ok(())
+    if chain_id_errors.is_empty() {
+        Ok(())
+    } else {
+        bail!(
```

Comment:
Could we document the expected response in the PR description and let the client teams give a :+1: ?